addons/account/models/product.py

```diff
@@ -269,6 +315,33 @@ def _compute_tax_string(self):
         for record in self:
             record.tax_string = record.product_tmpl_id._construct_tax_string(record.lst_price)
 
+    @api.model
+    def _get_products_and_most_recent_invoice_date(self, partner_id: int, journal_type: str, product_ids: list):
```

Comment:
There is no typing in the file. So don't start to type randomly and keep a coherence in the file

Reply:
Why use product_ids and @api.model while the method is on product itself

Reply:
I can also type all the parameters in this file if you prefer :nerd_face: 
Joking, I removed that and I removed the parameter too (I moved and modified multiple times this method, I forgot to adapt this part accordingly)